### ISSUE TITLE
FreeBSD >> operator does not append

### ISSUE THREAD

**Issue by benrubson** [vgough/encfs#334]
Hello,

I just discovered this :

```
$ uname -sr
FreeBSD 11.0-RELEASE-p8
$ /tmp/encfs --version
encfs version 1.9.1
$
$ echo -e "line1\nline2\nline3" > test.txt
$ cat test.txt 
line1
line2
line3
$ echo line4 >> test.txt 
$ cat test.txt 
line4
line2
line3
$
```

EncFS does not correctly append content to the end of the file, as if it was not able to seek.

Ben

**Comment by Aikhjarto** [comment]
Cannot confirm on Leap 42.2 with kernel 4.11.6 with encfs 1.9.1. 
On my system appending works.

**Comment by benrubson** [comment]
Thank you for your test @Aikhjarto.
What is strange is that the `internalModification` test correctly works.
https://github.com/vgough/encfs/blob/v1.9.1/tests/normal.t.pl#L86
So sounds like `seek()` performs correctly.

**Comment by benrubson** [comment]
Thank you for your test @Aikhjarto.
What is strange is that the `internalModification` test correctly works.
https://github.com/vgough/encfs/blob/v1.9.1/tests/normal.t.pl#L86
So sounds like `seek()` performs correctly.
Not sure exactly what `>>` internally does.

**Comment by benrubson** [comment]
One more test :
```
# cat test.pl 
unlink "perltest.txt";
open(my $out, ">", "perltest.txt");
print($out "line1\nline2\nline3\n");
close($out);
open($out, ">>", "perltest.txt");
print($out "line4\n");
close($out);
# 
# perl test.pl 
# cat perltest.txt 
line1
line2
line3
line4
#
```

So appending using Perl seems to work...
Really strange !

**Comment by benrubson** [comment]
OK, same issue using another Fuse FS, such as `sshfs`.

I've then opened a FreeBSD bug :
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=220185

**Comment by rfjakob** [comment]
Sounds like freebsd has a problem with files opened with the O_APPEND flags

**Comment by benrubson** [comment]
Good news, @cemeyer found the bug and proposed a FreeBSD patch 👍 
I then close this issue.
Many thx !

**Comment by cemeyer** [comment]
Perl must not use O_APPEND internally.  Happy to help.

**Comment by benrubson** [comment]
Good news, patch will be available in upcoming FreeBSD 11.1 :
https://svnweb.freebsd.org/changeset/base/320689
